src/commands/createTransactionRegisterDelegate.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * LiskHQ/lisky
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import {
+	getStdIn,
+	getPassphrase,
+	getFirstLineFromString,
+} from '../utils/input';
+import { createCommand } from '../utils/helpers';
+import commonOptions from '../utils/options';
+import transactions from '../utils/transactions';
+
+const description = `Creates a transaction which will register a delegate for an existing account if broadcast to the network.
+
+	Examples:
+	- create transaction register delegate
+	- create transaction 2
+`;
+
+const createDelegate = username => ([passphrase, secondPassphrase]) =>
+	transactions.createDelegate(passphrase, username, secondPassphrase);
+
+export const actionCreator = vorpal => async ({ username, options }) => {
+	const {
+		passphrase: passphraseSource,
+		'second-passphrase': secondPassphraseSource,
+	} = options;
+
+	return getStdIn({
+		passphraseIsRequired: passphraseSource === 'stdin',
+		dataIsRequired: secondPassphraseSource === 'stdin',
+	})
+		.then(stdIn => getPassphrase(vorpal, passphraseSource, stdIn.passphrase, { shouldRepeat: true })
+			.then(passphrase => (secondPassphraseSource ? getPassphrase(
+				vorpal,
+				secondPassphraseSource,
+				getFirstLineFromString(stdIn.data),
+				{ shouldRepeat: true, displayName: 'your second secret passphrase' },
+			) : Promise.resolve(null))
+				.then(secondPassphrase => [passphrase, secondPassphrase]),
+			),
+		)
```

Comment:
Will the above always be exactly the same for different transaction creators? We should probably have a helper function to avoid this boilerplate.

Reply:
Okay, this will then be handled here https://github.com/LiskHQ/lisky/issues/318